app/scripts/modules/core/application/config/trafficGuard/trafficGuardConfig.component.ts

```diff
@@ -0,0 +1,87 @@
+import {module, toJson} from 'angular';
+import {cloneDeep} from 'lodash';
+
+import {
+  ACCOUNT_SERVICE, AccountService, IAccountDetails, IRegion,
+  IAggregatedAccounts
+} from 'core/account/account.service';
+import {Application} from 'core/application/application.model';
+import {IViewState} from '../footer/configSectionFooter.component';
+import {TRAFFIC_GUARD_CONFIG_HELP} from './trafficGuardConfig.help';
+
+interface ITrafficGuard {
+  account: string;
+  location: string;
+  stack: string;
+  detail: string;
+}
+
+export class TrafficGuardConfigController {
+
+  public application: Application;
+  public accounts: IAccountDetails[] = [];
+  public regionsByAccount: { [account: string]: string[] };
+  public config: ITrafficGuard[];
+  public initializing: boolean = true;
+
+  public viewState: IViewState = {
+    originalConfig: null,
+    originalStringVal: null,
+    saving: false,
+    saveError: false,
+    isDirty: false,
+  };
+
+  static get $inject() { return ['accountService']; }
+
+  public constructor(private accountService: AccountService) {}
+
+  public $onInit(): void {
+    if (this.application.notFound) {
+      return;
+    }
+    this.config = this.application.attributes.trafficGuards || [];
+    this.viewState.originalConfig = cloneDeep(this.config);
+    this.viewState.originalStringVal = toJson(this.viewState.originalConfig);
```

Comment:
What do you think about starting to defer to native implementations where they exist?  E.g., for here, using `JSON.parse` instead.  It will assist in the transition away from ng1.